Graphql checkout tutorial (#4271)
magento/devdocs: guides/v2.3/graphql/tutorials/checkout/checkout-shopping-cart.md

---
layout: tutorial
group: graphql

> title: Step 2. Create empty cart

Reviewer: ```suggestion
title: Step 2. Create an empty cart
```

Author: **fixed**

subtitle: GraphQL checkout tutorial
level3_subgroup: graphql-checkout
return_to:
  title: GraphQL Overview
  url: graphql/index.html
menu_order: 20
functional_areas:
  - Integration
contributor_name: Atwix
contributor_link: https://www.atwix.com/
---

**Request**

> The following mutation query creates an empty cart:

Reviewer: ```suggestion
The following mutation creates an empty cart:
```

Author: **fixed**

```text
mutation {
  createEmptyCart
}
```

**Response**

```json
{
  "data": {
    "createEmptyCart": "A7jCcOmUjjCh7MxDIzu1SeqdqETqEa5h"
  }
}
```

> In the next tutorial steps the unique shopping cart identifier - `A7jCcOmUjjCh7MxDIzu1SeqdqETqEa5h` will be mentioned as `{{ CART_ID }}`.

Reviewer: ```suggestion
In the subsequent tutorial steps, the unique shopping cart identifier `A7jCcOmUjjCh7MxDIzu1SeqdqETqEa5h` will be listed as `{{ CART_ID }}`.
```

Author: **implemented**

{:.bs-callout .bs-callout-info}

> Send customer's authorization token in the `Authorization` parameter of the header if you create a shopping cart for a registered customer. See ["Get customer authorization token"]({{ page.baseurl }}/graphql/get-customer-authorization-token.html) to get more details.

Reviewer: Use the same edited paragraph.

Author: > Use the same edited paragraph.

@keharper , could you please clarify what exactly I should fix here? Thank you!

Author: **done**

---
layout: tutorial
group: graphql
title: Step 2. Create an empty cart
subtitle: GraphQL checkout tutorial
level3_subgroup: graphql-checkout
return_to:
  title: GraphQL Overview
  url: graphql/index.html
menu_order: 20
functional_areas:
  - Integration
contributor_name: Atwix
contributor_link: https://www.atwix.com/
---

> 

Author: **done**

**Request**

The following mutation creates an empty cart:

```text
mutation {
  createEmptyCart
}
```

**Response**

```json
{
  "data": {
    "createEmptyCart": "A7jCcOmUjjCh7MxDIzu1SeqdqETqEa5h"
  }
}
```

In the subsequent tutorial steps, the unique shopping cart identifier `A7jCcOmUjjCh7MxDIzu1SeqdqETqEa5h` will be listed as `{{ CART_ID }}`.

> {:.bs-callout .bs-callout-info}

Author: **done**

For logged-in customers, send the customer's authorization token in the `Authorization` parameter of the header. See ["Get customer authorization token"]({{ page.baseurl }}/graphql/get-customer-authorization-token.html) for more information.

> ### Verify this step {#verify-step}

Author: **done**

> There are no additional verification steps. `quote`.`entity_id` value is not displayed on the website or in Admin.

Author: **done**
